### PR TITLE
Made tomography requirements optional, documented their installation

### LINKED ISSUE #122: Document installation of tomography dependencies.

### PR DESCRIPTION
**Pull request by ntezak**
This fixes #122 and #121

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -11,18 +11,23 @@ setenv =
...
 deps=
     numpy
-    cython==0.24.1
 commands=
     pip install -r requirements.txt
 
+    ; check docs build correctly without complicated deps
+    sphinx-build -b html docs/ docs/_build
+
+    pip install cython==0.24.1
```
Comment (ampolloreno) [inline tox.ini]: Can we just call `pip install -r optional-requirements.py` here?

**Comment by ampolloreno** [comment]
@ntezak It looks like the build is failing because the solver is not functional for python2, but it is for python3. This happened to me, and I fixed it by installing the libblas and lapack libraries, or installing `python-dev`. There might be a race condition where something is installing while tox starts for the first test... not quite sure.

**Comment by ntezak** [comment]
@ampolloreno Huh, maybe installing `python-dev` in the semaphore config will help, that's a good idea. I accidentally removed your apt-get command last week so I was not sure what was all in it

**Comment by ntezak** [comment]
@ampolloreno you rock. This fixed the build. I will address/fix your other comments by EOD

File: grove/tomography/tomography.py [hunk 2]
```diff
@@ -48,6 +47,20 @@
 from grove.tomography.utils import QI, QX, QY, bitlist_to_int, sample_assignment_probs
 
 _log = logging.getLogger(__name__)
+
```
[the remark this answers is not in the document]
Comment (ntezak) [reply grove/tomography/tomography.py]: Yeah I will try to clean this up

File: README.md [hunk 3]
```diff
@@ -47,6 +47,8 @@ Dependencies
 * Matplotlib (useful for plotting)
 * pytest (optional, for development testing)
 * mock (optional, for development testing)
+* cvxpy (optional, for tomography, see below)
+* qutip (optional, for tomography, see below)
```
[the remark this answers is not in the document]
Comment (ntezak) [reply README.md]: I'm ok just removing that list

File: tox.ini (hunk 1 above)
Comment (ampolloreno) [reply tox.ini]: :+1:

File: optional-requirements.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+cython
```
[the remark this answers is not in the document]
Comment (ampolloreno) [reply optional-requirements.txt]: :+1:

File: grove/tomography/utils.py [hunk 5]
```diff
@@ -44,6 +41,45 @@
...
 _log = logging.getLogger(__name__)
 
+
+_QUTIP_ERROR_LOGGED = False
+
+
+def import_qutip():
+    """
+    Try importing the qutip module, log an error if unsuccessful.
+    :return:
```
Comment (ampolloreno) [inline grove/tomography/utils.py]: Incomplete docstring, and missing newline.

File: grove/tomography/utils.py [hunk 6]
```diff
@@ -44,6 +41,45 @@
...
+    return qutip
+
+
+_CVXPY_ERROR_LOGGED = False
+
+
+def import_cvxpy():
+    """
+    Try importing the qutip module, log an error if unsuccessful.
+    :return:
```
Comment (ampolloreno) [inline grove/tomography/utils.py]: Incomplete docstring, and missing newline.

**Comment by ntezak** [comment]
Boom, please re-review @ampolloreno

**Pull request merged**